src/ecdsa/ecdh.py

```diff
@@ -80,25 +87,39 @@ def _get_shared_secret(self, remote_public_key):
     def set_curve(self, key_curve):
         """
         Set working curve for ecdh operation.
+
         :param key_curve: curve from `curves`
+        :type key_curve: Curve
         :return: none
         """
         self.curve = key_curve
 
     def generate_private_key(self):
         """
         Generate local private key for ecdh operation with curve that was set.
-        :return: public (verifying) key from this private key
+
+        :raises NoCurveError: Curve must be set before key generation.
+
+        :return: public (verifying) key from this private key.
+        :rtype: byte string
```

Comment:
`self.private_key.get_verifying_key()` returns a byte string?

Reply:
i thought to return bytes but better will be VerifyingKey object